### PR TITLE
Refactor/redirect to last page - PELAGOS-3673

### PR DESCRIPTION
**Pull request by mickel1138**

### CODE REVIEW

**Comment by praneethpr** [comment]
> Everything almost works as expected, but because we treat dif a little differently it seems to not redirect from there.

Dataset submission also doesn't redirect

**Comment by mickel1138** [comment]
PELAGOS-3673

File: src/Pelagos/Bundle/AppBundle/Controller/SecurityController.php [hunk 1]
```diff
@@ -13,21 +15,25 @@
...
      * The login action.
      *
+     * @param Request $request The request object.
+     *
      * @Route("/login", name="security_login")
      *
      * @return Response A Response instance.
      */
-    public function loginAction()
+    public function loginAction(Request $request)
```
Comment (praneethpr) [inline src/Pelagos/Bundle/AppBundle/Controller/SecurityController.php]: Can remove Request variable now

File: src/Pelagos/Bundle/AppBundle/Form/LoginForm.php [hunk 2]
```diff
@@ -5,6 +5,8 @@
 use Symfony\Component\Form\AbstractType;
 use Symfony\Component\Form\Extension\Core\Type\PasswordType;
 use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\Extension\Core\Type\HiddenType;
```
Comment (praneethpr) [inline src/Pelagos/Bundle/AppBundle/Form/LoginForm.php]: Can remove unused hiddenType

File: src/Pelagos/Bundle/AppBundle/Controller/SecurityController.php [hunk 3]
```diff
@@ -5,6 +5,8 @@
 use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
 
 use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Request;
```
Comment (praneethpr) [inline src/Pelagos/Bundle/AppBundle/Controller/SecurityController.php]: Can remove this

File: src/Pelagos/Bundle/AppBundle/Controller/SecurityController.php [hunk 4]
```diff
@@ -13,21 +15,25 @@
  */
 class SecurityController extends Controller
 {
+    use TargetPathTrait;
```
Comment (praneethpr) [inline src/Pelagos/Bundle/AppBundle/Controller/SecurityController.php]: Can remove this

**Pull request merged**